src/ray/raylet/local_task_manager.cc

```diff
@@ -778,6 +781,31 @@ bool LocalTaskManager::CancelTask(
   return false;
 }
 
+void LocalTaskManager::CancelInfeasiblePlacementGroupTasks(
+    std::deque<std::shared_ptr<internal::Work>> &tasks) {
+  for (auto work_it = tasks.begin(); work_it != tasks.end();) {
+    if ((*work_it)->GetState() != internal::WorkStatus::WAITING) {
```

Comment:
Hmm, what will happen to other WorkStatus like WAIINT_FOR_WORKER when a pg is removed.